labelmodels/noisy_partial.py

```diff
@@ -0,0 +1,329 @@
+from .label_model import LabelModel, init_random
+import numpy as np
+from scipy import sparse
+import torch
+from torch import nn
+from copy import deepcopy as dc
+import logging
+
+
+'''
+Set Ops
+'''
+def intercect(l1, l2):
+    return [value for value in l1 if value in l2]
+
+def union(l1, l2):
+    return list(set(l1) | set(l2))
+
+
+
+class NPLMLearningConfig(object):
+    """Container for hyperparameters used by NPLM during learning"""
+
+    def __init__(self):
+        """Initializes all hyperparameters to default values"""
+        self.epochs = 200
+        self.batch_size = 8192
+        self.step_size = 0.1
+        self.step_schedule = 'p'
+        self.step_size_mult = 0.1
+        self.momentum = 0.8
+        self.random_seed = 0
+
+
+class NPLM(LabelModel):
+    """A generative label model that assumes that all partial labeling functions are
+    conditionally independent given the true class label/
+    """
+
+    def __init__(self, num_classes,
+                 fid2clusters,
+                 init_acc=0.7, preset_cb=None,
+                 learn_class_balance=True,
+                 device='cuda:0'):
```

Comment:
Thanks! Although it is default as GPU thru out, during the LM setup, it will check for CUDA availability and switch to CPU if GPU is not available but I will default it to CPU anyway to avoid any further confusion!.